### DISCUSSION TITLE
lookup的complex_selector

### DISCUSSION THREAD

**Discussion by mdj33** [z2o-k7e/halo2-step-by-step#48] [Q&A]
lookup 必须使用 complex_selector，因为 Halo2 可以根据这个标记知道这种 Selecotr 列不需要优化，而普通的 Selector 则可能会被 Layouter 进行合并等优化操作。

这里不是很理解， selector 的优化就是合并吗，如果complex_selector合并会有什么问题吗

**Comment by readygo67** [comment]
selector 的 优化方案可以参考:  https://zcash.github.io/halo2/design/implementation/selector-combining.html。简单的说就是将 k 个simple selector 合并成1个selector，当它的值等于m(m<k) 时,表示原来的第m个selector 被enable了。

**Reply by mdj33** [reply]
那请问为啥complex 不能一起优化呢
